From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Prathamesh Shete <pshete@nvidia.com>
Date: Thu, 18 Jul 2024 08:05:18 +0000
Subject: [PATCH 185/241] NVIDIA: SAUCE: gpio-tegra: add 1ms delay before
 setting debounce

First interrupt is missing when debounce is enabled
This happens because it takes 1ms for sampling to happen.
Hence add 1ms deay just before enabling debounce.
This is as per suggestions by HW team.

http://nvbugs/4733609

Signed-off-by: Prathamesh Shete <pshete@nvidia.com>
Change-Id: I06ef32ce40077efe845bfe553ceeaa2df29c26b7
(cherry picked from commit 0d08a7ebe1413f6411629f9ecbf50de97eaeab2d)
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3193513
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>
Tested-by: Petlozu Pravareshwar <petlozup@nvidia.com>
Reviewed-by: Petlozu Pravareshwar <petlozup@nvidia.com>

diff --git a/drivers/gpio/gpio-tegra186.c b/drivers/gpio/gpio-tegra186.c
index 420ca3e77b30..8309f9ae4a68 100644
--- a/drivers/gpio/gpio-tegra186.c
+++ b/drivers/gpio/gpio-tegra186.c
@@ -15,6 +15,7 @@
 #include <linux/platform_device.h>
 #include <linux/property.h>
 #include <linux/seq_file.h>
+#include <asm/delay.h>
 
 #include <dt-bindings/gpio/tegra186-gpio.h>
 #include <dt-bindings/gpio/tegra194-gpio.h>
@@ -446,6 +447,7 @@ static int tegra186_gpio_set_config(struct gpio_chip *chip,
 
 	debounce = DIV_ROUND_UP(debounce, USEC_PER_MSEC);
 
+	udelay(1000);
 	value = TEGRA186_GPIO_DEBOUNCE_CONTROL_THRESHOLD(debounce);
 	writel(value, base + TEGRA186_GPIO_DEBOUNCE_CONTROL);
 
-- 
2.43.0

